src/ensembl/src/shared/hooks/useOutsideClick.tsx

```diff
@@ -17,11 +17,23 @@
 import { useEffect } from 'react';
 
 export default function useOutsideClick<T extends HTMLElement>(
-  ref: React.RefObject<T>,
+  refs: React.RefObject<T>[],
```

Comment:
Why did you decide against allowing both a single ref and an array of refs?

Reply:
I thought about allowing both but this way I felt it will be consistent throughout. I also feel `refs` is slightly better than `refOrRefs`.

Reply:
>  it will be consistent throughout

But will it be convenient?

There are lots of libraries that allow passing different types of arguments. `classnames` is a good example — you can pass a single class name to it, or you can pass several class names, or you can pass an array of class names, or you can pass a configuration object. It's not forcing the user to be consistent — it's allowing them to pass whatever is more convenient at the moment.

I think passing a single ref if you only care about one ref, and passing an array of refs if you care about many refs is more convenient than having to always pass an array of refs, even when it's a single ref.

Reply:
Fair enough. Shall we simply call it `refs`?

Reply:
this check:

`const refs = Array.isArray(refOrRefs) ? refOrRefs : [refOrRefs];` will rename the parameter to refs, so that it can then conveniently be called `refs` within the body of the function :-)

Reply:
you can call `refOrRefs` `input` if you don't like the name `refOrRefs`  😄 

I just wanted to avoid calling something that may be singular with a name that is plural